src/libcore/num/mod.rs

```diff
@@ -2504,6 +2504,30 @@ impl fmt::Display for TryFromIntError {
     }
 }
 
+#[unstable(feature = "try_from", issue = "33417")]
+impl TryFrom<bool> for bool {
+    type Error = ();
+
+    fn try_from(u: bool) -> Result<bool, ()> {
+        Ok(u)
+    }
+}
+
+macro_rules! try_from_for_bool {
+    ($($source:ty),*) => {$(
+        #[unstable(feature = "try_from", issue = "33417")]
+        impl TryFrom<$source> for bool {
+            type Error = ();
+
+            fn try_from(u: $source) -> Result<bool, ()> {
+                Ok(u != 0)
```

Comment:
Is there precedent in Rust for this being the canonical way (or even "a" right way) of turning integers into bools? For example, there's no `as` cast from integers to bools, no `From` impl, no `num::NumCast`, etc.